# Financial Analytics: Lab 1

In [1]:
import pandas as pd
import scipy as sp
import numpy as np
import matplotlib.pyplot as plt
import pandas_datareader.data as getData
import pandas_datareader as pdr
import datetime
import numpy

#### Question 1

What is the present value of $418 received in 12 years with an annual discount rate of 2.2%?


In [2]:
PV_Q1 = sp.pv(0.022, 12, 0, 418, when = 'end')*-1
print("The Present value is " + "$" + "%.2f" % PV_Q1)
#sp.pv(rate,nper,pmt,fv=0,when='end')

The Present value is $321.93


#### Question 2

If you deposit $2,400 in the bank for 15 years and the annual interest rate us 1.75%, what is the future value of your deposit in 15 years?


In [3]:
#Future value calculation
FV_Q2 = sp.fv(0.0175, 15, 0, 2400,  when = 'end') * -1

print("Assuming there is only a single deposit of 2400 made on the first year, the Future value of the deposit will be, " + "$" + "%.2f" % FV_Q2)
#sp.fv(rate,nper,pmt,pv=0, when = 'end')

Assuming there is only a single deposit of 2400 made on the first year, the Future value of the deposit will be, $3113.35


#### Question 3

Annual payment of an investment is $35 with 18 years remaining in the investment vehicle. What is the present value if the annual discount rate is 4.32%, compounded semi-annually?


In [4]:
PV_Q3 = sp.pv(0.0432/2, 36, 35, 0,  when = 'end')*-1
print("The Present value of the Investment compounded semi-annually  is " + "$" + "%.2f" % PV_Q3)

The Present value of the Investment compounded semi-annually  is $869.61


#### Question 4

a)	If you are trying to achieve a savings goal of $5,500 by the end of 2023 (assuming you started Jan 1st, 2020), how much do you have to save each year if the corresponding annual rate is 2.98%?

b)	How much do you have to save semi-annually?


In [5]:
Yearly_payment = numpy.pmt(0.0298, 4, 5500, 0, when='end')*-1

print("The yearly savings should be " + "$" + "%.2f" % Yearly_payment)

The yearly savings should be $1478.94


In [7]:
Yearly_payment = numpy.pmt(0.0298/2, 2*4, 5500,0, when='end')*-1

print("The semi-annual savings should be " + "$" + "%.2f" % Yearly_payment)

The semi-annual savings should be $734.39


#### Question 5

A capital investment in an equipment with an upfront cost of $23,540 will provide you with the following annual cash flow stream (paid end of year):

1.	$2,000

2.	$1,456

3.	$3,230

4.	$6,850

5.	$2,384

6.	$1,234

7.	$5,987

8.	$4,190

The project will incur the following cost for maintenance and repair (paid end of year):

Year 3: ($2,984)

Year 4: ($1,837)

Year 6-8: ($2,000)

Calculate the NPV of the investment and comment on whether you should invest in the project. Why or why not? What is the IRR of the investment?

In [8]:
cashflow = [-23540,2000, 1456, (3230-2984),(6850-1837),2384,(1234-2000),(5987-2000),(4190-2000)]

In [9]:
NPV = sp.npv(0.087,cashflow)

In [10]:
print("The NPV of the investment for the equipment is " + "$" + "%.2f" % NPV)

The NPV of the investment for the equipment is $-12231.90


In [11]:
IRR = sp.irr(cashflow)

In [12]:
print("The IRR of the investment for the equipment is " + "$" + "%.2f" % IRR)

The IRR of the investment for the equipment is $-0.07


In [13]:
#verify that the calculated irr sets NPV to zero
NPV_irr = sp.npv(IRR,cashflow)

In [14]:
print("The calculated IRR sets the NPV to  " + "$" + "%.2f" % NPV_irr)

The calculated IRR sets the NPV to  $0.00


No, i would not invest in this project. The NPV is at -12231.90, therefore the cost exceeds the benefits. The IRR is also -0.07 which indicates that the present value of all cash inflows would not be able to cover the cost of capital. 

Since the project's rate of return will not outweigh its costs, the project should be rejected.